nrfx_config.h

```diff
@@ -48,6 +48,44 @@
 #define NRFX_CLOCK_ENABLED 1
 #endif
 
+#ifdef CONFIG_CLOCK_CONTROL_NRF_K32SRC_RC
+#if defined(CONFIG_SOC_SERIES_NRF91X) || defined(CONFIG_SOC_SERIES_NRF53X)
+#define NRFX_CLOCK_CONFIG_LF_SRC 1
+#else
+#define NRFX_CLOCK_CONFIG_LF_SRC 0
+#endif
+#elif CONFIG_CLOCK_CONTROL_NRF_K32SRC_XTAL
+#if defined(CONFIG_SOC_SERIES_NRF91X) || defined(CONFIG_SOC_SERIES_NRF53X)
+#define NRFX_CLOCK_CONFIG_LF_SRC 2
+#else
+#define NRFX_CLOCK_CONFIG_LF_SRC 1
+#endif
+#elif CONFIG_CLOCK_CONTROL_NRF_K32SRC_SYNTH
+#if defined(CONFIG_SOC_SERIES_NRF91X)
+#error "Configuration not supported"
```

Comment:
I also think we don't need such checks here. Let Kconfig handle these dependencies.